sdk/identity/azure_identity/src/process/mod.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore workdir
+
+use azure_core::{
+    credentials::AccessToken,
+    error::{Error, ErrorKind, Result},
+    process::Executor,
+};
+use std::{ffi::OsStr, marker::PhantomData, sync::Arc};
+
+use crate::env::Env;
+
+/// Runs a command in the appropriate platform shell and processes the output
+/// using the specified `OutputProcessor`.
+///
+/// - Windows: Runs `cmd /C {command}` in %SYSTEMROOT%
+/// - Everywhere else: Runs `/bin/sh -c {command}` in /bin
+pub(crate) async fn shell_exec<T: OutputProcessor>(
+    executor: Arc<dyn Executor>,
+    env: &Env,
+    command: &str,
+    _: PhantomData<T>,
+) -> Result<AccessToken> {
+    let (workdir, program, c_switch) = if cfg!(target_os = "windows") {
+        let system_root = env.var("SYSTEMROOT").map_err(|_| {
+            Error::message(
+                ErrorKind::Credential,
+                "SYSTEMROOT environment variable not set",
+            )
+        })?;
+        (system_root, "cmd", "/C")
+    } else {
+        ("/bin".to_string(), "/bin/sh", "-c")
+    };
+
+    let command_string = format!("cd {workdir} && {command}");
+    let args = vec![OsStr::new(c_switch), OsStr::new(&command_string)];
+
+    let status = executor.run(OsStr::new(program), &args).await;
+
+    match status {
+        Ok(output) if output.status.success() => {
+            T::deserialize_token(&String::from_utf8_lossy(&output.stdout))
+        }
+        Ok(output) => {
+            let stderr = String::from_utf8_lossy(&output.stderr);
+            let message = if let Some(error_message) = T::get_error_message(&stderr) {
+                error_message.to_string()
+            } else if output.status.code() == Some(127) || stderr.contains("' is not recognized") {
+                format!("{} not found on path", T::tool_name())
```

Comment:
One, I'd capitalize `PATH`. That's consistent with most other CLIs I've seen. But, I wouldn't use the display name of the tool. Why not just return the tool name, which we already have above anyway, e.g., `az` or `azd`? Make it plain as day.